src/data/lovelace/config/badge.ts

```diff
@@ -1,7 +1,7 @@
 import { Condition } from "../../../panels/lovelace/common/validate-condition";
 
 export interface LovelaceBadgeConfig {
-  type?: string;
+  type: string;
```

Comment:
There will still be a lot of configs around that won't have a `type`, we should deal with those so changing this type might not be wise?